### ISSUE TITLE
Add color picker popover to ColorInput

### ISSUE THREAD

**Issue by jmaen** [motion-canvas/motion-canvas#910]
**Description**
Add a color picker popover to the existing `ColorInput` component. One could then select a color with instant visual feedback instead of having to type the color explicitly, which I think is way more convenient. 
Of course, at the moment there is only really one usage of the `ColorInput` for the background color. However, I can imagine multiple use cases in the future, which would definitely benefit from a more powerful color picker.

**Proposed solution**
The picker could open when the `ColorPreview` component inside the `ColorInput` is clicked. 

I looked for existing solutions and found [react-colorful](https://github.com/omgovich/react-colorful), which I think looks promising.

**Comment by jmaen** [comment]
I played around with it a little just for fun, I imagine it could look something like that (sorry for the color artifacts):

![color-picker](https://github.com/motion-canvas/motion-canvas/assets/47495425/676a39e4-af04-4371-9388-0627fb9331bb)

**Comment by jmaen** [comment]
I'm sorry, I didn't want to close the issue..

**Comment by aarthificial** [comment]
While it's true that we'll need a color picker sooner or later for the sake of editable signals, I am strongly against using third-party components in the editor as they destroy any sense of consistency in terms of the visuals and the feel.

It's important to keep in mind that Motion Canvas is meant to feel like a desktop application and not a website.
Here's an example of the color picker from Blender:
![image](https://github.com/motion-canvas/motion-canvas/assets/64662184/ab8ca571-d97d-484c-9f90-68cc9a6cb797)
Most notably, it uses HSV sliders that both look and work the same as any other field in the application.

As such, the color picker should be implemented using our own components.
This shouldn't be too complex as it's just a few sliders and two CSS gradients on top of each other. 
We also already depend on `chroma-js` which provides all the logic necessary for color conversion an manipulation.

**Comment by jmaen** [comment]
I'll look into that. I guess the new `NumberInput` will come in handy for the sliders.

**Comment by jmaen** [comment]
Does it have to be the same layout as in Blender, i.e. one (circular) map for hue and saturation and another one for brightness, or would you also be fine with one map for saturation and brightness and another linear one for hue? I think the latter would be easier to implement and I like the easy way of keeping the saturation and brightness while playing around with the hue, but I guess that comes down to personal preference.

**Comment by aarthificial** [comment]
Yes, the square + hue slider from the gif is better. It's common in most drawing software so I think it would be a good default.

**Comment by jmaen** [comment]
I had some other things going on but I finally got to work on it again. The functionality is there now but before spending time on styling I wanted to clarify how exactly you imagine the component to look like.

That's the current (temporary) state:
![grafik](https://github.com/motion-canvas/motion-canvas/assets/47495425/04222896-7c5b-4f11-88e7-bf1083369311)

**Comment by aarthificial** [comment]
Looking good so far!
Here are some tweaks:
![image](https://github.com/motion-canvas/motion-canvas/assets/64662184/d1cd92f6-dd32-40a1-a172-7920f790afe2)
1. The vertical gap should be `8px` everywhere, and the padding should be `16px` to be consistent with the rest of the editor.
  Here's it on a `4x4px` grid:
![image](https://github.com/motion-canvas/motion-canvas/assets/64662184/09c0ba3b-3d84-415e-9365-f35e03da3634)
2. Selection circles should be `12x12px` with the one over the hue being black to stand out better.
3. Values should be presented with 4 decimal places
4. Inputs should have nested labels on the left
5. There should be an alpha/opacity slider at the bottom.
6. Hue should range from `0` to `1`

**Comment by jmaen** [comment]
Sounds good, got everything looking good now. Just one more thing, where do you want the popover to be located relative to the color preview? Just some fixed distance to the right? If it just opens where the click happens, the background blends with the rest of the settings pane. 
Or do you want to make the popover background darker, like Blender handles it?

**Comment by aarthificial** [comment]
Having it fixed on the right is ok for now, 12px to match the padding around the zoom picker.
Tho I'm also considering making it collapsible, similarly to how colors in the inspector are presented:
![image](https://github.com/motion-canvas/motion-canvas/assets/64662184/76990819-ca2d-49ab-99d3-dac72e676080)
I think in the future this would work better with the inspector on the right
